src/content/docs/en/guides/environment-variables.mdx

```diff
@@ -71,7 +71,7 @@ const isDev = import.meta.env.DEV;
 ### `.env` files
 Environment variables can be loaded from `.env` files in your project directory.
 
-You can also attach a mode (either `production` or `development`) to the filename, like `.env.production` or `.env.development`, which makes the environment variables only take effect in that mode.
+You can also attach a mode to the filename, like `.env.production`, `.env.development`, or `.env.staging`, which makes the environment variables only take effect in that mode. The mode can be changed by passing the [`--mode` flag](/en/reference/cli-reference/#--mode-string) to the `astro dev` or `astro build` commands (defaults to `"development"` and `"production"` respectively).
```

Comment:
``````suggestion
You can also add `.production`, `.development` or a custom mode name to the filename itself (e.g `env.testing`, `.env.staging`). This allows you to use different sets of environment variables at different times.

The `astro dev` and `astro build` commands default to `"development"` and `"production"` modes, respectively.  You can run these commands with the [`--mode` flag](#--mode-string) to pass a different value for `mode` and load the matching `.env` file.

This allows you to run the dev server or build your site connecting to different APIs:

```bash
# Run the dev server connected to a "staging" API
astro dev --mode staging

# Build a site that connects to a "production" API with additional debug information
astro build --devOutput

# Build a site that connects to a "testing" API
astro build --mode testing
```
``````
I really liked what you had in the changeset, Bjorn! What about something like this here?

ALSO, I think this entire section should go AFTER we show the basic .env file, that is the simple use case. I would put this after the example, and before the "read more"

Reply:
Looks great! I was afraid that it's getting a bit long, but if we put that in a separate section then it's probably fine then. I'll commit this change and re-sort it.